### PR TITLE
can read config from cli / env / toml & docker for arm64 & support docker run --rm ceresdb --help

### PR DESCRIPTION
**Pull request by xxaier**
1. docker for arm64
3. support docker run --rm ceresdb --help
4. can read config from cli / env / toml

image
image

### CODE REVIEW

File: src/bin/ceresdb-server.rs [hunk 1]
```diff
@@ -58,15 +58,33 @@ fn main() {
                 .takes_value(true)
                 .help("Set configuration file, eg: \"/path/server.toml\""),
         )
+        .arg(
```
Comment (jiacai2050) [inline src/bin/ceresdb-server.rs:61]: This method seems awkward for me, doesn't the original issue is to support env vars to set config?

File: .github/workflows/dockerhub-publish-nightly.yml [hunk 2]
```diff
@@ -38,8 +40,9 @@ jobs:
             echo "BUILD_DATE=$(TZ=':Asia/Shanghai' date '+%Y%m%d')" >> $GITHUB_ENV
             echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
       - name: Build and Push Docker Image
-        uses: docker/build-push-action@v3
+        uses: docker/build-push-action@v4
         with:
+          platforms: linux/amd64,linux/arm64
```
Comment (jiacai2050) [inline .github/workflows/dockerhub-publish-nightly.yml:45]: Have you tride build on linux/arm64?

File: src/bin/ceresdb-server.rs [hunk 3]
```diff
@@ -58,15 +58,33 @@ fn main() {
...
         .get_matches();
 
-    let mut config = match matches.value_of("config") {
-        Some(path) => {
-            let mut toml_buf = String::new();
-            toml::parse_toml_from_path(path, &mut toml_buf).expect("Failed to parse config.")
-        }
-        None => Config::default(),
-    };
+    // let toml = cli_env_toml::cli_env_toml_str(
```
Comment (jiacai2050) [inline src/bin/ceresdb-server.rs:71]: Delete this.

File: src/bin/ceresdb-server.rs (hunk 1 above)
Comment (xxaier) [reply src/bin/ceresdb-server.rs:61]: 这个环境变量的配置方式是参考 rust config库的配置方式  参见 https://blog.logrocket.com/configuration-management-in-rust-web-services/
image

比如 qdrant 就是用的这种配置方式，如下图
https://github.com/qdrant/qdrant
image


这个配置方式的优点是，不需要修改原来的代码且通用，不然每个地方都要加一个判断，很麻烦

File: .github/workflows/dockerhub-publish-nightly.yml (hunk 2 above)
Comment (xxaier) [reply .github/workflows/dockerhub-publish-nightly.yml:45]: 可以构建，但是没测试，我觉得可以先添加构建，再去全面测试下

File: src/bin/ceresdb-server.rs (hunk 1 above)
Comment (jiacai2050) [reply src/bin/ceresdb-server.rs:61]: You can cli_env_toml to do the heavy task, and this is a very new crate, so I'm concern with stability.

Also, although set config one by one may seems dumb at first, but it make code simple and easy to read, so I suggest you use original style to add env support.

I think following 4 envs are enough for your case.:
```
CERESDB_BIND_ADDR
CERESDB_HTTP_PORT
CERESDB_MYSQL_PORT
CERESDB_GRPC_PORT
```

File: .github/workflows/dockerhub-publish-nightly.yml (hunk 2 above)
Comment (jiacai2050) [reply .github/workflows/dockerhub-publish-nightly.yml:45]: How about apply it only in nightly image?

If everything goes fine for some period, we can apply it to production image.

**Comment by jiacai2050** [comment]
@xxaier Hi, do you have free time to push this PR forward?

If not, I can do some minor changes based on your branch, Thanks.

**Comment by ShiKaiWi** [comment]
Inactive for too long. If any updates, feel free to reopen it.

**Pull request closed without merging**